src/dynimage.rs

```diff
@@ -279,14 +328,73 @@ impl DynamicImage {
         }
     }
 
-    /// Return this image's pixels as a byte vector.
-    pub fn raw_pixels(&self) -> Vec<u8> {
-        image_to_bytes(self)
+    /// Return a reference to an 16bit RGB image
+    pub fn as_rgb16(&self) -> Option<&Rgb16Image> {
+        match *self {
+            DynamicImage::ImageRgb16(ref p) => Some(p),
+            _ => None,
+        }
+    }
+
+    /// Return a mutable reference to an 16bit RGB image
+    pub fn as_mut_rgb16(&mut self) -> Option<&mut Rgb16Image> {
+        match *self {
+            DynamicImage::ImageRgb16(ref mut p) => Some(p),
+            _ => None,
+        }
+    }
+
+    /// Return a reference to an 16bit RGBA image
+    pub fn as_rgba16(&self) -> Option<&Rgba16Image> {
+        match *self {
+            DynamicImage::ImageRgba16(ref p) => Some(p),
+            _ => None,
+        }
+    }
+
+    /// Return a mutable reference to an 16bit RGBA image
+    pub fn as_mut_rgba16(&mut self) -> Option<&mut Rgba16Image> {
+        match *self {
+            DynamicImage::ImageRgba16(ref mut p) => Some(p),
+            _ => None,
+        }
+    }
+
+    /// Return a reference to an 16bit Grayscale image
+    pub fn as_luma16(&self) -> Option<&Gray16Image> {
+        match *self {
+            DynamicImage::ImageLuma16(ref p) => Some(p),
+            _ => None,
+        }
+    }
+
+    /// Return a mutable reference to an 16bit Grayscale image
+    pub fn as_mut_luma16(&mut self) -> Option<&mut Gray16Image> {
+        match *self {
+            DynamicImage::ImageLuma16(ref mut p) => Some(p),
+            _ => None,
+        }
+    }
+
+    /// Return a reference to an 16bit Grayscale image with an alpha channel
+    pub fn as_luma_alpha16(&self) -> Option<&GrayAlpha16Image> {
+        match *self {
+            DynamicImage::ImageLumaA16(ref p) => Some(p),
+            _ => None,
+        }
+    }
+
+    /// Return a mutable reference to an 16bit Grayscale image with an alpha channel
+    pub fn as_mut_luma_alpha16(&mut self) -> Option<&mut GrayAlpha16Image> {
+        match *self {
+            DynamicImage::ImageLumaA16(ref mut p) => Some(p),
+            _ => None,
+        }
     }
 
-    /// Return a view on the raw sample buffer.
-    pub fn as_flat_samples(&self) -> FlatSamples<&[u8]> {
```

Comment:
Could it instead be changed to return an `Option` here? I always saw it as a more generic version of the `as_rgb8(&self)` bunch of methods and that return type would fit in nicely.

Reply:
There are now separate methods with optional returns for `u8` and `u16`.